ha/util/consul_kv_store.py

```diff
@@ -122,7 +145,16 @@ def set(self, key: str, val: str=None):
         self._verify_data(key)
         if self.key_exists(key):
             raise Exception(f"Key {key} already exists in kv store.")
-        self._consul.kv.put(self._prepare_key(key), val)
+        k = self._prepare_key(key)
+
+        if self._enable_batch_put:
+            Log.debug(f"[consul-op] Putting key value in cache for key: {k}")
+            self._payload[k] = val
```

Comment:
This is not very good way to implement, functions need to be more explicit with respect to the parameters that they accept. Flags don't help much with the readability. Better to have an explicit function, e.g. `commit_batch()` that accepts a list of Key-values.

Reply:
Yes, i.e. good same discussed with Neerav also,

but in ha_setup() we are not directly adding keys we are generating events that get proceeded by the event manager and during the process of each event it generates and adds required key(s), so at no point in time you will have a bunch of keys to add so I have taken an approach to catch is in kV store class without changing the code event manager otherwise it will be design change for the event manager and may need to think more.

Yes, we can write a separate function commit_batch() and put call the keys directly from ha_setup() and don't process events through the event manager.
But needs to understand and discuss if it is possible and what are the consequences of skipping event processing otherwise we land up adding too much code in ha_setup and maybe it is duplicating much node code.

Hi @mssawant @shaileshsvc, @ArchanaLimaye, @NeeravChoudhari,
Needs to discuss whether is above change is feasible now and its consequences. otherwise can schedule later.
Need input on this.

Reply:
@mukhtar-inamdar, See, anyway its the HA that is generating the key on receiving the event. Is it that HA doesn't know before hand what type of events it will receive?

Reply:
Hi @mssawant,
CC @ArchanaLimaye 
Below are the points from the discussion in the group:

Yes, actually we are initialing HA through events, so that can be skipped, but needs to understand the consequences.

So just add keys with default values and once the deployment is done, start processing upcoming events from the K8s monitor will look like work, but needs to evaluate that keys with the default value, can be just hardcoded or needs to generate by some code,

this may take the effort of story points 3 to 5. @ArchanaLimaye  your thoughts, please?

Reply:
will fix in: https://jts.seagate.com/browse/CORTX-31319